cvat-sdk/cvat_sdk/core/downloading.py

```diff
@@ -27,21 +29,52 @@ class Downloader:
     def __init__(self, client: Client):
         self._client = client
 
+    @staticmethod
+    def _get_server_filename(response: urllib3.HTTPResponse) -> str:
+        content_disposition = next(
+            (
+                filename
+                for part in response.headers.get("Content-Disposition", "").split(";")
+                if (filename := part.strip()) and filename.startswith("filename=")
+            ),
+            None,
+        )
+
+        filename = None
+        if content_disposition:
+            filename_parts = content_disposition.split("=", maxsplit=1)
+            if len(filename_parts) == 2:
+                filename = filename_parts[1].strip('"')
+                filename = Path(filename).name
+
+        if not filename:
+            raise Exception(
+                "Can't find the output filename in the server response, "
+                "please try to specify the output filename explicitly"
+            )
+
+        return filename
```

Comment:
I think we need to at least check that the filename is not one of the [Windows reserved names](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions), like `nul`, `con`, etc. Otherwise, a malicious server could cause unintended effects on the client machine by using one of these names.

Reply:
Added validation